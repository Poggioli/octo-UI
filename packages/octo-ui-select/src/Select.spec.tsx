import { render } from "@testing-library/react";
import {
  Select,
  SelectContent,
  SelectGroup,
  SelectItem,
  SelectLabelGroup,
  SelectSeparator
} from "./Select";

describe("<Select />", () => {
  it(`GIVEN a <Select />
      WHEN render is called
      AND has selected value
      THEN should render a <Select />`, () => {
    const rendered = render(
      <Select value="apple" placeholder="Select a food">
        <SelectContent>
          <SelectGroup>
            <SelectLabelGroup>Fruits</SelectLabelGroup>
            <SelectItem value="apple">Apple</SelectItem>
            <SelectItem disabled value="banana">
              Banana
            </SelectItem>
          </SelectGroup>
          <SelectSeparator />
          <SelectGroup>
            <SelectLabelGroup>Vegetables</SelectLabelGroup>
            <SelectItem value="lettuce">Lettuce</SelectItem>
          </SelectGroup>
        </SelectContent>
      </Select>
    );
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <Select />
      WHEN render is called
      AND has selected value
      AND it's disabled
      THEN should render a <Select />`, () => {
    const rendered = render(
      <Select disabled value="apple" placeholder="Select a food">
        <SelectContent>
          <SelectGroup>
            <SelectLabelGroup>Fruits</SelectLabelGroup>
            <SelectItem value="apple">Apple</SelectItem>
            <SelectItem disabled value="banana">
              Banana
            </SelectItem>
          </SelectGroup>
          <SelectSeparator />
          <SelectGroup>
            <SelectLabelGroup>Vegetables</SelectLabelGroup>
            <SelectItem value="lettuce">Lettuce</SelectItem>
          </SelectGroup>
        </SelectContent>
      </Select>
    );
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <Select />
      WHEN render is called
      AND it's disabled
      THEN should render a <Select />`, () => {
    const rendered = render(
      <Select disabled placeholder="Select a food">
        <SelectContent>
          <SelectGroup>
            <SelectLabelGroup>Fruits</SelectLabelGroup>
            <SelectItem value="apple">Apple</SelectItem>
            <SelectItem disabled value="banana">
              Banana
            </SelectItem>
          </SelectGroup>
          <SelectSeparator />
          <SelectGroup>
            <SelectLabelGroup>Vegetables</SelectLabelGroup>
            <SelectItem value="lettuce">Lettuce</SelectItem>
          </SelectGroup>
        </SelectContent>
      </Select>
    );
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <Select />
      WHEN render is called
      THEN should render a <Select />`, () => {
    const rendered = render(
      <Select placeholder="Select a food">
        <SelectContent>
          <SelectGroup>
            <SelectLabelGroup>Fruits</SelectLabelGroup>
            <SelectItem value="apple">Apple</SelectItem>
            <SelectItem disabled value="banana">
              Banana
            </SelectItem>
          </SelectGroup>
          <SelectSeparator />
          <SelectGroup>
            <SelectLabelGroup>Vegetables</SelectLabelGroup>
            <SelectItem value="lettuce">Lettuce</SelectItem>
          </SelectGroup>
        </SelectContent>
      </Select>
    );
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <Select />
      WHEN render is called
      AND it's openned
      THEN should render a <Select />`, () => {
    const rendered = render(
      <Select defaultOpen value="apple" placeholder="Select a food">
        <SelectContent>
          <SelectGroup>
            <SelectLabelGroup>Fruits</SelectLabelGroup>
            <SelectItem value="apple">Apple</SelectItem>
            <SelectItem disabled value="banana">
              Banana
            </SelectItem>
          </SelectGroup>
          <SelectSeparator />
          <SelectGroup>
            <SelectLabelGroup>Vegetables</SelectLabelGroup>
            <SelectItem value="lettuce">Lettuce</SelectItem>
          </SelectGroup>
        </SelectContent>
      </Select>
    );
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <SelectGroup />
      WHEN toString is called
      THEN return a string`, () => {
    expect(SelectGroup.toString()).toBe(".octo-UI-c-PJLV");
  });

  it(`GIVEN a <SelectItem />
      WHEN toString is called
      THEN return a string`, () => {
    expect(SelectItem.toString()).toBe(".octo-UI-c-gmtBZL");
  });

  it(`GIVEN a <SelectLabelGroup />
      WHEN toString is called
      THEN return a string`, () => {
    expect(SelectLabelGroup.toString()).toBe(".octo-UI-c-gKmfoP");
  });

  it(`GIVEN a <Select />
      WHEN toString is called
      THEN return a string`, () => {
    expect(Select.toString()).toBe(".octo-UI-c-PJLV");
  });

  it(`GIVEN a <SelectSeparator />
      WHEN toString is called
      THEN return a string`, () => {
    expect(SelectSeparator.toString()).toBe(".octo-UI-c-bkmZcV");
  });

  it(`GIVEN a <SelectContent />
      WHEN toString is called
      THEN return a string`, () => {
    expect(SelectContent.toString()).toBe(".octo-UI-c-cjYrgy");
  });
});
