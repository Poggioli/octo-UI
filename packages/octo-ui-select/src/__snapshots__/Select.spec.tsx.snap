// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> GIVEN a <Select />
      WHEN render is called
      AND has selected value
      AND it's disabled
      THEN should render a <Select /> 1`] = `
<div>
  <button
    aria-autocomplete="none"
    aria-controls="radix-:r6:"
    aria-expanded="false"
    class="octo-UI-c-eTCyLC octo-UI-c-eTCyLC-hBbGqh-size-standard octo-UI-c-eTCyLC-fGbOAu-color-unstyled"
    data-disabled=""
    data-state="closed"
    dir="ltr"
    disabled=""
    role="combobox"
    type="button"
  >
    <span
      style="pointer-events: none;"
    >
      Apple
    </span>
    <span
      aria-hidden="true"
      class="octo-UI-c-PJLV"
    >
      <svg
        fill="none"
        height="15"
        viewBox="0 0 15 15"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`<Select /> GIVEN a <Select />
      WHEN render is called
      AND has selected value
      THEN should render a <Select /> 1`] = `
<div>
  <button
    aria-autocomplete="none"
    aria-controls="radix-:r0:"
    aria-expanded="false"
    class="octo-UI-c-eTCyLC octo-UI-c-eTCyLC-hBbGqh-size-standard octo-UI-c-eTCyLC-fGbOAu-color-unstyled"
    data-state="closed"
    dir="ltr"
    role="combobox"
    type="button"
  >
    <span
      style="pointer-events: none;"
    >
      Apple
    </span>
    <span
      aria-hidden="true"
      class="octo-UI-c-PJLV"
    >
      <svg
        fill="none"
        height="15"
        viewBox="0 0 15 15"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`<Select /> GIVEN a <Select />
      WHEN render is called
      AND it's disabled
      THEN should render a <Select /> 1`] = `
<div>
  <button
    aria-autocomplete="none"
    aria-controls="radix-:rc:"
    aria-expanded="false"
    class="octo-UI-c-eTCyLC octo-UI-c-eTCyLC-hBbGqh-size-standard octo-UI-c-eTCyLC-fGbOAu-color-unstyled"
    data-disabled=""
    data-placeholder=""
    data-state="closed"
    dir="ltr"
    disabled=""
    role="combobox"
    type="button"
  >
    <span
      style="pointer-events: none;"
    >
      Select a food
    </span>
    <span
      aria-hidden="true"
      class="octo-UI-c-PJLV"
    >
      <svg
        fill="none"
        height="15"
        viewBox="0 0 15 15"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`<Select /> GIVEN a <Select />
      WHEN render is called
      AND it's openned
      THEN should render a <Select /> 1`] = `
<div
  aria-hidden="true"
  data-aria-hidden="true"
>
  <button
    aria-autocomplete="none"
    aria-controls="radix-:ro:"
    aria-expanded="true"
    class="octo-UI-c-eTCyLC octo-UI-c-eTCyLC-hBbGqh-size-standard octo-UI-c-eTCyLC-fGbOAu-color-unstyled"
    data-state="open"
    dir="ltr"
    role="combobox"
    type="button"
  >
    <span
      style="pointer-events: none;"
    >
      Apple
    </span>
    <span
      aria-hidden="true"
      class="octo-UI-c-PJLV"
    >
      <svg
        fill="none"
        height="15"
        viewBox="0 0 15 15"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`<Select /> GIVEN a <Select />
      WHEN render is called
      THEN should render a <Select /> 1`] = `
<div>
  <button
    aria-autocomplete="none"
    aria-controls="radix-:ri:"
    aria-expanded="false"
    class="octo-UI-c-eTCyLC octo-UI-c-eTCyLC-hBbGqh-size-standard octo-UI-c-eTCyLC-fGbOAu-color-unstyled"
    data-placeholder=""
    data-state="closed"
    dir="ltr"
    role="combobox"
    type="button"
  >
    <span
      style="pointer-events: none;"
    >
      Select a food
    </span>
    <span
      aria-hidden="true"
      class="octo-UI-c-PJLV"
    >
      <svg
        fill="none"
        height="15"
        viewBox="0 0 15 15"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </span>
  </button>
</div>
`;
